ooniprobe/View/Settings/SettingsCategoriesTableViewController.m

```diff
@@ -46,11 +48,38 @@ - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath
     if ([current isEqualToString:@"about_ooni"]){
         [self performSegueWithIdentifier:current sender:self];
     }
+    else if ([current isEqualToString:@"send_email"]){
+        [self sendEmail];
+    }
     else
         [self performSegueWithIdentifier:@"toSettings" sender:self];
     [tableView deselectRowAtIndexPath:indexPath animated:YES];
 }
 
+-(void)sendEmail{
+    if ([MFMailComposeViewController canSendMail]){
+        MFMailComposeViewController *mailer = [[MFMailComposeViewController alloc] init];
+        mailer.mailComposeDelegate = self;
+        [mailer setSubject:[NSString stringWithFormat:@"[bug-report] OONI Probe iOS %@", [[[NSBundle mainBundle] infoDictionary] valueForKey:@"CFBundleShortVersionString"]]];
+        NSArray *toRecipients = [NSArray arrayWithObjects:@"bugs@openobservatory.org", nil];
+        [mailer setToRecipients:toRecipients];
+        NSString *device = [NSString stringWithFormat:@"\n\n\napp_version %@\ndevice_model %@\nos_version %@", [[[NSBundle mainBundle] infoDictionary] valueForKey:@"CFBundleShortVersionString"], [UIDevice currentDevice].modelIdentifier, [UIDevice currentDevice].systemVersion];
+        [mailer setMessageBody:device isHTML:NO];
+        [[mailer navigationBar] setTranslucent:NO];
+        [[mailer navigationBar] setTintColor:[UIColor colorWithRGBHexString:color_white alpha:1.0f]];
+        [[mailer navigationBar] setTitleTextAttributes: [NSDictionary dictionaryWithObjectsAndKeys: [UIColor colorWithRGBHexString:color_white alpha:1.0f], NSForegroundColorAttributeName, [UIFont fontWithName:@"FiraSans-SemiBold" size:16], NSFontAttributeName, nil]];
+        [self presentViewController:mailer animated:YES completion:nil];
```

Comment:
Where are you using the new class?

Reply:
[UIDevice currentDevice].modelIdentifier

Reply:
What I don't get is why you're using the parent class and not the derived class. I guess this has to do with my little understanding of ObjectiveC, so an explanation will help me to know this language better and be more useful in the future.

Reply:
The new class has all the method of the basic class+ the new ones.
By including the new class in the header all the methods to UIDevice will be called on the new class. Don't really know how but it's how it works